### PR TITLE
Remove check current state membership up to date

### PR DESCRIPTION
**Pull request by Fizzadar**
Bit of cleanup; originally added [back in 2019](https://github.com/matrix-org/synapse/pull/5706) with schema compat changing since, so should be safe to assume this is complete.

Contributed by Nick @ Beeper (@fizzadar).

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog).
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by reivilibre** [comment]
The problem is that it's still valid for someone to upgrade from whatever version in 2019 directly to the latest version, so in general it's not safe to assume that it's complete :-(.
To address that, you'd have to make the background job run as a schema delta (which blocks startup) instead.

I can't remember the exact recipe, but part of me seems to remember it involves retroactively updating the old schema delta which scheduled the background job to instead do it immediately (not in the background).

That said, maybe the right approach is actually to write a new schema delta file (probably in Python code https://matrix-org.github.io/synapse/latest/development/database_schema.html#python-delta-files) that:

- checks to see if (what was) the background job is still scheduled to run
- if so, does the work in the foreground and unschedules the background job — this is hopefully not much more complicated than copying the current background job and just doing it in the foreground a loop.

**Comment by Fizzadar** [comment]
>    That said, maybe the right approach is actually to write a new schema delta file (probably in Python code https://matrix-org.github.io/synapse/latest/development/database_schema.html#python-delta-files) that

Agreed I think this is a nicer approach; have done so in https://github.com/matrix-org/synapse/pull/13745/commits/80e5d3b40822b92efd94072bc92993051bb1ce38.

File: synapse/storage/databases/main/roommember.py [hunk 1]
```diff
@@ -346,60 +288,32 @@ def _get_room_summary_txn(
...
-                    FROM room_memberships as m
-                    INNER JOIN current_state_events as c USING (room_id, event_id)
-                    WHERE c.type = 'm.room.member' AND c.room_id = ?
-                    ORDER BY
-                        CASE m.membership WHEN ? THEN 1 WHEN ? THEN 2 ELSE 3 END ASC,
-                        c.event_id ASC
-                    LIMIT ?
-                """
+            # Note, rejected events will have a null membership field, so
+            # we we manually filter them out.
```
Comment (squahtx) [inline synapse/storage/databases/main/roommember.py:306]: The old comment still applies
```suggestion
            # we order by membership and then fairly arbitrarily by event_id so
            # heroes are consistent
            # Note, rejected events will have a null membership field, so
            # we we manually filter them out.
```

File: synapse/storage/databases/main/roommember.py [hunk 2]
```diff
@@ -639,30 +553,15 @@ async def get_rooms_for_user_with_stream_ordering(
...
-                SELECT room_id, e.instance_name, e.stream_ordering
-                FROM current_state_events AS c
-                INNER JOIN room_memberships AS m USING (room_id, event_id)
-                INNER JOIN events AS e USING (room_id, event_id)
-                WHERE
-                    c.type = 'm.room.member'
-                    AND c.state_key = ?
-                    AND m.membership = ?
-            """
+        sql = """
```
Comment (squahtx) [inline synapse/storage/databases/main/roommember.py:556]: The comment looks like it's still relevant

```suggestion
        # We use `current_state_events` here and not `local_current_membership`
        # as a) this gets called with remote users and b) this only gets called
        # for rooms the server is participating in.
        sql = """
```

**Review by squahtx — COMMENTED** [review]
Looks good! Some very minor comments.

**Review by squahtx — APPROVED** [review]

**Pull request merged**